server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -5148,6 +5148,7 @@ public PhysicalNetworkTrafficType addTrafficTypeToPhysicalNetwork(Long physicalN
             if (TrafficType.Public.equals(trafficType)) {
                 List<String> isolationMethods = network.getIsolationMethods();
                 if ((isolationMethods.size() == 1 && isolationMethods.get(0).toLowerCase().equals("vxlan"))
+                        // TODO: what happens if there are multiple isolation methods, why are multiple isolation methods allowed per pnet? They seem to be filterable by provider/preifx (PhysicalNetwork.java). Documentation says only two are allowed but code and tests contradict this.
```

Comment:
In theory a physical network represents a network device on the phy. host; Something say a eth0 or a bond0 (if it's a team/bond) or a bridge0 (if it's bridge on the hypervisor host), but then a phy network could carry tagged VLAN traffic such as eth0.10, or bridge0-10 carrying VLAN 10 etc. For the same phy. network, a shared network with a different VLAN can be created.

I don't have the historical context to answer why it's the way it is, generally we don't expect more than a single isolation method of a phy network - but some network plugins can decide to do things differently for example support both VLAN, GRE etc. so it's possible the isolate methods is a list. For the sake of the issue you're trying to address, the scope is limited to returning data that's already in the database.